akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -885,6 +891,29 @@ object Source {
     Source.fromGraph(
       new QueueSource(bufferSize, overflowStrategy, maxConcurrentOffers).withAttributes(DefaultAttributes.queueSource))
 
+  /**
+   * Creates a `Source` that is materialized as an [[akka.stream.DroppingQueueSource]].
+   * You can push elements to the queue and they will be emitted to the stream if there is demand from downstream,
+   * otherwise they will be buffered until request for demand is received. The buffer size is passed in as a parameter.
+   * Elements in the buffer will be discarded if downstream is terminated.
+   *
+   * Pushed elements may be dropped if there is no space available in the buffer. Elements will also be dropped if the
+   * queue is completed or failed or if the stream completes.
+   *
+   * Acknowledgement of pushed elements is immediate.
+   * [[akka.stream.DroppingQueueSource.offer]] returns [[akka.stream.QueueOfferResult]] which is implemented as:
+   *
+   * `QueueOfferResult.Enqueued`       element was added to buffer
+   * `QueueOfferResult.Dropped`        element was dropped
+   * `QueueOfferResult.QueueComplete`  the queue was completed with [[akka.stream.DroppingQueueSource.complete]]
+   * `QueueOfferResult.Failure`        the queue was failed with [[akka.stream.DroppingQueueSource.fail]] or if the
+   *                                   stream failed
+   *
+   * @param bufferSize size of buffer in element count
+   */
+  def droppingQueue[T](bufferSize: Int): Source[T, DroppingQueueSource[T]] =
+    Source.fromGraph(new DroppingSourceQueueStage[T](bufferSize))
```

Comment:
Sorry, I haven't been able to review this PR yet, but I think `boundedQueue` would be a better name.

Even better would be to keep it as `Source.queue` overload, but maybe we are using same parameters?

Reply:
> I think boundedQueue would be a better name

SGTM

> Even better would be to keep it as `Source.queue` overload, but maybe we are using same parameters?

We could use the overload `def queue(bufferSize: Int): Source[T, BoundedQueueSource[T]]`.  It will have a different return type than the other overloads, as well as a different implementation.  

We could discuss deprecating the other overloads and supporting the back-pressure (and maybe the fail) `OverflowStrategy` as separate `Source` factory methods.

Reply:
I renamed the impl. to `BoundedQueue`.

I'm now using the `Source.queue(bufferSize: Int)` overload for the new impl and swapping the impl. when the `OverflowStrategy.dropNew` strategy is used in the other overloads.